### PR TITLE
:bug: Fix color-picker recent colors

### PR DESCRIPTION
**Pull request by pabloalba**
Fixes reopened https://tree.taiga.io/project/penpot/issue/4013

### CODE REVIEW

File: frontend/src/app/main/ui/workspace/colorpicker.cljs [hunk 1]
```diff
@@ -61,8 +61,17 @@
         set-harmony-tab!    (mf/use-fn #(reset! active-tab :harmony))
         set-hsva-tab!       (mf/use-fn #(reset! active-tab :hsva))
 
+        drag?               (mf/use-state false)
+
         handle-change-color
-        (mf/use-fn #(st/emit! (dc/update-colorpicker-color %)))
+        (mf/use-fn
+         (mf/deps drag?)
```
Comment (niwinz) [inline frontend/src/app/main/ui/workspace/colorpicker.cljs]: the deps should include the value of drag?, it should be `@drag?` here

File: frontend/src/app/main/ui/workspace/colorpicker.cljs [hunk 2]
```diff
@@ -96,14 +105,19 @@
         on-activate-radial-gradient
         (mf/use-fn #(st/emit! (dc/activate-colorpicker-gradient :radial-gradient)))
 
+        on-start-drag
+        (mf/use-fn
+         (mf/deps state)
```
Comment (niwinz) [inline frontend/src/app/main/ui/workspace/colorpicker.cljs]: looks like you don't need state here, so this dep can de removed

File: frontend/src/app/main/ui/workspace/colorpicker.cljs [hunk 3]
```diff
@@ -96,14 +105,19 @@
...
+         (fn []
+           (reset! drag? true)
+           (st/emit! (dwu/start-undo-transaction))))
+
         on-finish-drag
         (mf/use-fn
          (mf/deps state)
```
Comment (niwinz) [inline frontend/src/app/main/ui/workspace/colorpicker.cljs]: the same here

File: frontend/src/app/main/ui/workspace/colorpicker.cljs (hunk 1 above)
Comment (pabloalba) [reply frontend/src/app/main/ui/workspace/colorpicker.cljs]: fixed

File: frontend/src/app/main/ui/workspace/colorpicker.cljs (hunk 2 above)
Comment (pabloalba) [reply frontend/src/app/main/ui/workspace/colorpicker.cljs]: fixed

File: frontend/src/app/main/ui/workspace/colorpicker.cljs (hunk 3 above)
Comment (pabloalba) [reply frontend/src/app/main/ui/workspace/colorpicker.cljs]: fixed

**Pull request merged**